### PR TITLE
Completed Labs 1&2, Assignment 1

### PR DESCRIPTION
**Pull request by elioc1341**
## What changes are you trying to make? (e.g. Adding or removing code, refactoring existing code, adding reports)
- Uploading completed Lab 1, Lab 2, and Assignment 1 Jupyter Notebook

## What did you learn from the changes you have made?
- The first and foremost is that neural networks are a completely new concept to me, so going through the labs and assignment slowly helped me build an understanding of how they work, and how tweaking different dials or parameters or even methods can help improve model performance.
- Further to the previous point, just actually typing out the code helped build practice and confidence in writing the syntax to complete the required tasks.
- I wasn’t fully able to follow what loss and accuracy means during class at first, but by going through the results slowly I can understand why these are important metrics when reviewing model results:
     Loss being the discrepancy between the prediction and actual results. The lower the number, the better.
     Accuracy being how many of the predictions were correct. The higher the number, the better.
- This was also the first hands-on approach were I would train models on images - previously in the previous modules, classifications were strictly using numerical/empirical data, so it was really interesting to see how the models would work when working with images instead.

## Was there another approach you were thinking about making? If so, what approach(es) were you thinking of?
- In part 4, there were many options for hyperparameters to test, such as kernels or number of layers or the L2 regularization. Any of these could have been done to complete the assignment.

## Were there any challenges? If so, what issue(s) did you face? How did you overcome it?
There were several challenges for these tasks:
- A major roadblock for Assignment 1 was I was getting the Kernel crashing error consistently when trying to fit the model. I initially spent time looking online trying to resolve the issue, as well as reopening VS code and restarting my computer. I eventually looked on the slack help channel to try troubleshooting. The initial steps for one thread were not working, but another suggested to run it in Google Colab and download the completed notebook. This eventually helped me continue working on the assignment, but it runs much slower than compared to local.
- Some smaller challenges were getting the Convolutional Neural Network and Dropout to work, as these are new things to me. I had to look up the documentation on keras to understand the arguments, and then try a few times to get the syntax working. I initially wasn’t too sure which arguments to play around with, but just rereading the assignment and also trying to keep it simple helped with completing them.
- Also, adding a second hidden layer in Lab 2 took me a bit longer to figure out the syntax. Logically, it makes sense each layer would feed into the next, to reach into the output. I just kept having to debug and fix the syntax so that my variables/suffixes flowed properly (X → H1 → H2 → O)

## How were these changes tested?
- I would individually run the blocks of code in the Jupyter notebook to see the results, or Google Colab for assignment 1. 
- I would also specifically try rerunning code to see how different results would show up when randomness is involved, such a model results, or even the images from the MNIST datasets.

## A reference to a related issue in your repository (if applicable)
N/A

## Checklist
- [x] I can confirm that my changes are working as intended

### CODE REVIEW

**Review by tianyi21 — APPROVED** [review]
Hi Eliot,

Thank you for submitting your DL week 1 assessments.

Your A1, Lab 1, and Lab 2 are now complete.

Thanks,
Tianyi [LS]
